### ISSUE TITLE
code doesn't work with another certificate issuer

### ISSUE THREAD

**Issue by myousry35** [bassemAgmi/EInvoicingSigner#23]
hi
i tried your code with Egypt Trust Sealing CA works fine but with MCDR 2019 doesn't  work and the code throw an exception 
A certificate chain could not be built to a trusted root authority.

**Comment by socertis** [comment]
yeah, that's because of the manufacturer of this certificate differs from Egypt Trust Sealing CA.
try to download the driver and point the code to the adequate .dll file.

**Comment by mrkindy** [comment]
@myousry35 did you try to change `TokenCertificate` to `MCDR CA` ?
are you use epass2003 ?

**Comment by myousry35** [comment]
yes i use epass2003 the code works well with all egypt trust ca token but with MCDR no . it makes an exception with the trusted root

**Comment by mrkindy** [comment]
did you try to change `TokenCertificate` to `MCDR CA` ?

**Comment by myousry35** [comment]
i included this option 
signer.IncludeOption = X509IncludeOption.EndCertOnly; and it signs the invoice but haven't tried yet to send a signed invoice with a MCDR token

**Comment by myousry35** [comment]
i find the cetificate by serial no
var foundCerts = store.Certificates.Find(X509FindType.FindBySerialNumber, certificate_serialno, false);

**Comment by mrkindy** [comment]
Well I think there is an issue with MCDR signed certificate you can use a new environment "New Windows Installation" and install the new MCDR Token

I think there is some conflict in the root certificate issued by Itida may EgyTrust uses a different certificate than MCDR

**Comment by myousry35** [comment]
this means a 2 windows pc for each certificate issuer

**Comment by mrkindy** [comment]
Try it first, it can solve your problem

**Comment by mrkindy** [comment]
And you can try to remove the Egypt Root certification and install the new one in MCDR Package

**Comment by myousry35** [comment]
that's why i had to get 2 pc for every certificate issuer to solve this root certificate problem exception i will try it thanks a lot for your help

**Comment by mrkindy** [comment]
You are welcome 😊,
When you try, please let me know  if this solution solved it or not

**Comment by socertis** [comment]
there is no need to do this, the root certificate is just a default option for windows.
but you can list all the available certificates on your system and select the one with issuer MCDR.
try to make use of Visual Studio's debugger to facilitate this attempt.
